torch/csrc/api/src/data/samplers/distributed.cpp

```diff
@@ -0,0 +1,164 @@
+#include <torch/data/samplers/distributed.h>
+#include <torch/serialize/archive.h>
+#include <torch/types.h>
+
+#include <algorithm>
+#include <cstddef>
+#include <random>
+#include <vector>
+
+namespace torch {
+namespace data {
+namespace samplers {
+
+DistributedRandomSampler::DistributedRandomSampler(
+    size_t size,
+    size_t num_replicas,
+    size_t rank,
+    bool allow_duplicates)
+    : DistributedSampler(size, num_replicas, rank, allow_duplicates),
+      begin_index_(0),
+      end_index_(0),
+      sample_index_(0) {
+  populate_indices();
```

Comment:
`reset()` now does `populate_indices()` too

Reply:
It had `populate_indices()` before, but only if the new_size is different. Since we want the epoch behavior to be the same, i.e. epoch 3 should produce the same sequence as epoch 1, 2, 3, we need to either sort the indices or populate them again. Populating is less costly.

Reply:
I agree, my point is that the call in the constructor is pointless.

Reply:
Yes, I removed it.